lib/adapter.js

```diff
@@ -4866,6 +4927,25 @@ function Adapter(options) {
         });
     };
 
+    /**
+     * This method update the cached values in this.usernames
+     *
+     * @returns {Promise<void>}
+     */
+    const updateUsernameCache = async () => {
+        // make sure cache is cleared
+        this.usernames = {};
+        try {
+            // get all users
+            const obj = await this.getObjectListAsync({startkey: 'system.user.', endkey: 'system.user.\u9999'});
+            for (const row of obj.rows) {
+                this.usernames[row.value.common.name] = {id: row.id};
```

Comment:
Can we trust that the name is always defined? Do we need emoty/null/undefined fallbacks?!

Reply:
Hm probably not, this it’s currently always called inside try catch because I wanted to inform about this error, but an error in early iterations will stop the execution of this method and thus it may be more desirable to log an error inside the function if non existing property name.